dbms/src/Storages/Transaction/TiKVKeyValue.h

```diff
@@ -46,6 +46,11 @@ struct StringObject : std::string
         : Base(str)
     {}
     static StringObject copyFrom(const Base & str) { return StringObject(str); }
+    template <bool B>
+    static StringObject<B> copyFrom(const StringObject<B> & other)
+    {
+        return StringObject(other.data(), other.dataSize());
+    }
```

Comment:
Do we really need to add this function? It accepts generating TiKVValue by copyFrom TiKVKey, which could be dangerous.

We can define it without the extra `bool B`. Or just calling `copyFrom(obj.getStr())`
```
    static StringObject copyFrom(const StringObject & other)
    {
        return StringObject(other.data(), other.dataSize());
    }
```

Reply:
Oh, I tried that even define it without the extra `bool B`, we can still copyFrom TiKVKey to TiKVValue :(

Reply:
I get your point, let me try some sfinae magic